classes/key_manager.php

```diff
@@ -0,0 +1,144 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Access key manager.
+ *
+ * @package    tool_pdfpages
+ * @author     Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright  2021 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_pdfpages;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Access key manager.
+ *
+ * @package    tool_pdfpages
+ * @author     Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright  2021 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class key_manager {
+
+    /**
+     * Create a user key.
+     *
+     * @param int $instanceid the instance ID to create key for.
```

Comment:
instanceid is a bit confusing. It seems like you are trying to match core create_user_key function, but there it is $instance and not $instanceid. What is it actually mean? Is it just a unique string?

Reply:
Renamed to `instance` in line with core APIs